src/audiomixerboard.cpp

```diff
@@ -63,13 +63,15 @@ CChannelFader::CChannelFader ( QWidget* pNW ) :
 
     // define the popup menu for the group checkbox
     pGroupPopupMenu = new QMenu ( "", pcbGroup );
-    pGroupPopupMenu->addAction ( tr ( "&No grouping" ), this, SLOT ( OnGroupMenuGrpNone() ) );
-    pGroupPopupMenu->addAction ( tr ( "Assign to group" ) + " &1", this, SLOT ( OnGroupMenuGrp1() ) );
-    pGroupPopupMenu->addAction ( tr ( "Assign to group" ) + " &2", this, SLOT ( OnGroupMenuGrp2() ) );
-    pGroupPopupMenu->addAction ( tr ( "Assign to group" ) + " &3", this, SLOT ( OnGroupMenuGrp3() ) );
-    pGroupPopupMenu->addAction ( tr ( "Assign to group" ) + " &4", this, SLOT ( OnGroupMenuGrp4() ) );
-#if ( MAX_NUM_FADER_GROUPS != 4 )
-# error "MAX_NUM_FADER_GROUPS must be set to 4, see implementation in CChannelFader()"
+    pGroupPopupMenu->addAction ( tr ( "&No grouping" ),
+        this, [=] { OnGroupMenuGrp ( INVALID_INDEX ); } );
+    for ( int iGrp = 0 ; iGrp < MAX_NUM_FADER_GROUPS ; iGrp++ )
+    {
+        pGroupPopupMenu->addAction ( tr ( "Assign to group" ) + ( QString ( " &%1" ) .arg( iGrp + 1 ) ),
+            this, [=] { OnGroupMenuGrp ( iGrp ); } );
+    }
+#if ( MAX_NUM_FADER_GROUPS != 8 )
+# error "MAX_NUM_FADER_GROUPS must be set to 8, see implementation in CChannelFader()"
```

Comment:
Is this necessary? The implementation looks like it works with any value of `MAX_NUM_FADER_GROUPS` up to 8. So suggest to only reject values > 8.

Reply:
Well, it was `!=` before, I just changed the number :).  But yes, it could say `>`.